src/test/java/RecursiveTest.java

```diff
@@ -0,0 +1,31 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RecursiveTest {
+    private static final int MILLION = 1_000_000;
```

Comment:
we usually call variables according to their purpose and not exact value 
```suggestion
    private static final int EXPECTED_SIZE or smt like this = 1_000_000;
```